enhancements/installer/gcp-ipi-shared-vpc.md

```diff
@@ -0,0 +1,262 @@
+
+A new parameter will be added to the install-config to specify the project id
+of the network and subnets. This parameter will then be used where necessary to
+validate existing network project resources as well as provision additional
+resources using these network resources. The default will be an empty string
+and will indicate these resources (should) exist in the default cluster
+project.
+
+#### installconfig.platform.gcp.privateManagedZone
+
+A new parameter will be added to the install-config to specify the private DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional private DNS records. The default
+will be an empty string, which will indicate these resources are to be created
+and managed in the project identified by the networkProjectID. When this
+parameter is set to “none”, the cluster will not be configured to manage
+private DNS resources.
+
+#### installconfig.platform.gcp.publicManagedZone
+
+A new parameter will be added to the install-config to specify the public DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional public DNS records. The default
+will be an empty string, which will indicate these resources are to be created
+and managed in the project identified by the networkProjectID (when publish is
+configured to “External”). When this parameter is set to “none”, the cluster
+will not be configured to manage public DNS resources.
+
+#### installconfig.platform.gcp.apiAddress
+
+A new parameter will be added to the install-config to specify a pre-existing
+compute address resource to be used when creating and managing the API load
+balancer(s).
+
+#### installconfig.platform.gcp.ingressAddress
+
+A new parameter will be added to the install-config to specify a pre-existing
+compute address resource to be used when creating and managing the default
+ingress load balancer(s).
+
+### Implementation Details/Notes/Constraints
+
+#### VPC, Network, and Subnetworks
+
+Shared VPC network and subnet(s) are to be created and configured in advance.
+They can then be configured in the install config using the following
+parameters as enabled by prior work. A new parameter (networkProjectID) will be
+added to specify in which project they exist. This parameter will be consumed
+by the Terraform templates as well as cluster manifests in order to configure
+the cluster to use these resources.
+
+- installconfig.platform.gcp.computeSubnet
+- installconfig.platform.gcp.controlPlaneSubnet
+- installconfig.platform.gcp.network
+
+#### Service Accounts
+
+Service Account(s) are to be created and configured in advance. These accounts
+must have additional permissions granted in the Host Project. The specific
+permissions depend on the desired level of automation. The install config's
+credentialsMode parameter must be set to passthrough so the cluster uses these
```

Comment:
1. The installer should default this to passthrough once we know this is XPN
2. a nit: but manual mode should be able to work here as well

Reply:
That is something to think about, for sure. One of the things I've realized is that many of the changes we will be making are useful even when not using XPN. Therefore, I'm trying to make sure I think about each knob with this in mind. We might be able to default to passthrough when networkProjectID is not empty. On the other hand, having the user specifically set this to passthrough makes it clear to them that the credentials they are using will become the cluster credentials. I believe we've had parallel conversations before.

Reply:
Can we enumerate the permissions changes/additional permissions that might be needed somewhere within the enhancement?

Reply:
In my testing, I used the following access in the host project. However, I suspect we can tune this down to individual requirements, and for each "thing" we are enabling (such as DNS, firewall, etc). I can add them to the enhancement if you think that would be useful, but I was hoping to scope them out better as part of the effort(s) to add these features.
Compute Instance Admin (beta)
Compute Network Admin
Compute Security Admin
DNS Administrator

Reply:
> I can add them to the enhancement if you think that would be useful

In my experience, this kind of information is useful to have in the enhancements so that product and docs teams have an awareness. Can always add this list provisionally now and update later if the list changes or becomes clearer in its definition

Reply:
Added permissions as requested.